web-frame/src/app/app-routing-error-handler.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * @license
+ * Copyright (c) 2018-2020 THREEANGLE SOFTWARE SOLUTIONS SRL
+ * Available under MIT license webFrame/LICENSE
+ */
+import { HttpErrorResponse } from '@angular/common/http';
+import { NavigationExtras, Router } from '@angular/router';
+import { trimStart } from 'lodash';
+import { ENVIRONMENT } from '~app-shared/config';
+import { AccessDeniedError, HttpStatusCode, PAGE_URL, PageNotFoundError, WebFrameErrorHandlerService } from '~app-shared/core';
+
+export function appRoutingErrorHandler(this: Router, err: unknown): unknown {
+  const extras: NavigationExtras = { skipLocationChange: true };
+  if (err instanceof AccessDeniedError) {
+    return this.navigateByUrl(PAGE_URL.ACCESS_DENIED_PAGE, extras);
+  } else if (err instanceof PageNotFoundError) {
+    return this.navigateByUrl(PAGE_URL.PAGE_NOT_FOUND_PAGE, extras);
+  } else if (err instanceof HttpErrorResponse) {
+    if (err.status === HttpStatusCode.FORBIDDEN) {
```

Comment:
This function is hard for the eyes to read, I would extract the block of code for HttpErrorReponse into a function HttpErrorResponseHandler. This way we can also reduce the nesting level by 1

Reply:
Fixed: https://github.com/3angleTech/webFrame/pull/140#discussion_r439669337